### PR TITLE
[FindInJarFiles] Show tree view only after first call

### LINKED ISSUE #812: Find In files view always shows
**Describe the bug**

Even if suer is not using `find in files` it will show up on the left. 

**To Reproduce** Steps to reproduce the behavior:

Look on the tabs view

**Expected behavior**

Tab is shown only when in use.

**Screenshots**

image

**Installation:**

- Operating system: Linux
- VSCode version: 1.61
- VSCode extension version: 1.10.15
- Metals version: (found in VSCode settings, under `metals.serverVersion`) 010.9

**Search terms**

search in file tab

### PR DESCRIPTION
**Pull request by dos65**
Fixes #812

### CODE REVIEW

File: src/findinfiles.ts [hunk 1]
```diff
@@ -53,6 +54,7 @@ export function createFindInFilesTreeView(
   provider: FindInFilesProvider,
   context: ExtensionContext
 ): TreeView<unknown> {
+  commands.executeCommand("setContext", "metals.showFindInFiles", false);
```
Comment (tgodzik) [inline src/findinfiles.ts:57]: It doesn't seem to cause it to disappear when we move to a separate window.

We should also do:
```
const onDidChangeVisibility = treeView.onDidChangeVisibility(async (e) => {
    if (!e.visible)
      commands.executeCommand("setContext", "metals.showFindInFiles", false);
  });
 context.subscriptions.push(onDidChangeVisibility);
```

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: src/findinfiles.ts (hunk 1 above)
Comment (tgodzik) [reply src/findinfiles.ts:57]: OR actually, maybe we want it to show still :thinking:

**Review by tgodzik — APPROVED** [review]
Actually, ignore me, this is exactly how references work.

**Comment by dos65** [comment]
> Actually, ignore me, this is exactly how references work.

Yep, I remember that I had some version of what you proposed but then compared with how default search and references and decided not to do that.

**Pull request merged**